src/controllers/multer.js

```diff
@@ -0,0 +1,32 @@
+const multer = require('multer')
+const path = require('path');
+
+const DIR = './public/'
+
+const storage = multer.diskStorage({
+  destination: function (req, file, cb) {
+    cb(null, DIR)
+  },
+  filename: function (req, file, cb) {
+    const uniqueSuffix = Date.now() + '-' + Math.round(Math.random() * 1E9 ) + path.extname(file.originalname)
+    cb(null, file.fieldname + '-' + uniqueSuffix)
+  }
+})
+
+
+const upload = multer({
+  storage: storage,
+  fileFilter: (req, file, cb) => {
+
+    if (file.mimetype === 'image/png' || file.mimetype === 'image/jpeg' || file.mimetype === 'image/jpeg') {
+      cb(null, true)
```

Comment:
Correct me if I'm wrong, but I don't see anywhere where the filename is saved to the DB to be accessed later. How is the image attached to the pet? Do we need a model to save the filename in the DB?

Reply:
Just tested it out and it looks like it's persistent (maybe without using the DB?) Really cool! Could you walk me through how that works? I'm confused haha

Reply:
haha! yeah it's a bit confusing. Here is what the code is doing.
1. In `ImageController.js`, ``upload.single('petimage')` saves the image in the `public` folder
2. In `module.exports.send`, the post request returns `filepath` (url) to the frontend
3. Frontend pass the `filepath` (url) to the `pet: picture` props and save the path along with the other pet's data

I am not sure if I descried well. Let me know if you have any questions!

Reply:
Ah I see, that's really clever! Looking in the DB now, it seems like the entire path is being saved, including localhost:8000. I think we should just save the relative path and then add in the HOSTNAME variable when searching for the image

Reply:
Thanks! I agree, the saving the relative path is much better. I changed it!